packages/cli/src/frameworks/nextjs/index.ts

```diff
@@ -144,45 +174,34 @@ export async function detectPagesOrAppDir(path: string): Promise<"pages" | "app"
   return "pages";
 }
 
-async function createTriggerPageRoute(
-  path: string,
-  endpointSlug: string,
-  isTypescriptProject: boolean,
-  pathAlias: string | undefined
-) {
-  const templatesDir = pathModule.join(templatesPath(), "nextjs");
-  const fileExtension = isTypescriptProject ? ".ts" : ".js";
-
-  //pages/api/trigger.js or src/pages/api/trigger.js
-  const apiRoutePath = pathModule.join(path, "pages", "api", `trigger${fileExtension}`);
-  const apiRouteResult = await createFileFromTemplate({
-    templatePath: pathModule.join(templatesDir, "pagesApiRoute.js"),
-    replacements: {
-      routePathPrefix: pathAlias ? pathAlias + "/" : "../../",
-    },
-    outputPath: apiRoutePath,
-  });
-  if (!apiRouteResult.success) {
-    throw new Error("Failed to create API route file");
+export async function detectNextVersion(path: string) {
+  const packageJsonContent = await readPackageJson(path);
+  if (!packageJsonContent) {
+    return null;
   }
-  logger.success(`✔ Created API route at ${apiRoutePath}`);
 
-  await createJobsAndTriggerFile(path, endpointSlug, fileExtension, pathAlias, templatesDir);
+  const versionNumberPattern = /[\d.]+|latest/;
+  if (packageJsonContent.dependencies?.next !== undefined)
+    return packageJsonContent.dependencies?.next?.match(versionNumberPattern)![0];
```

Comment:
There is a risk here that the user has a weird value for their version and we get null. Then this will throw an exception.

I'm going to change this to this:

```suggestion
if (packageJsonContent.dependencies?.next !== undefined)
    return packageJsonContent.dependencies?.next?.match(versionNumberPattern)?.at(0) ?? null;
  if (packageJsonContent.devDependencies?.next !== undefined)
    return packageJsonContent.devDependencies?.next?.match(versionNumberPattern)?.at(0) ?? null;
```